code/backend/getactivecore/src/main/java/com/bu/getactivecore/service/users/UsersService.java

```diff
@@ -100,10 +100,12 @@ public LoginResponseDto loginUser(LoginRequestDto requestDto) {
         Authentication authentication = m_authManager.authenticate(
                 new UsernamePasswordAuthenticationToken(requestDto.getUsername(), requestDto.getPassword()));
 
+        Optional<Users> existingUser =m_userRepo.findByUsername(requestDto.getUsername());
+
         // If authentication is successful, the user is logged in
-        if (authentication.isAuthenticated()) {
+        if (authentication.isAuthenticated() && existingUser.isPresent()) {
             String token = m_jwtApi.generateToken(requestDto.getUsername());
-            return new LoginResponseDto(token);
+            return new LoginResponseDto(token, existingUser.get().getUserId(), existingUser.get().getUsername(), existingUser.get().getEmail());
```

Comment:
When `authenticated()` returns true, that means the requested user exists in our database. This also means the current context has the user, so no need to make a database hit, we can do the following:

```suggestion
        if (authentication.isAuthenticated()) {
            String token = m_jwtApi.generateToken(requestDto.getUsername());
            Users user = ((UserPrincipal) authentication.getPrincipal()).getUser();
            return new LoginResponseDto(token, user.getUserId(), user.getUsername(), user.getEmail());

```


I made the mistake of providing the `Users` entity to `UserPrincipal`, it will be addressed in the back registration ticket.